src/clr-core/internal/utils/framework.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+// DO NOT USE for conditionally applying behavior based on the host framework.
+// These utilities are only for detecting host frameworks for debugging and logging.
+// Host framework specific code should exist outside of the @clr/core package.
```

Comment:
I want to make it clear here we do not change logical behavior in Core for external frameworks. This is to help our logging tool for Github issues. `window.CDS.getVersion()`. Right now we can only get the active version for Angular. React and Vue we can detect but currently do not expose the running version.